x-pack/packages/kbn-elastic-assistant/impl/assistant/use_conversation/index.tsx

```diff
@@ -51,299 +48,264 @@ interface CreateConversationProps {
 }
 
 interface SetApiConfigProps {
-  conversationId: string;
-  apiConfig: Conversation['apiConfig'];
-}
-
-interface SetConversationProps {
   conversation: Conversation;
+  apiConfig: Conversation['apiConfig'];
 }
 
 interface UseConversation {
-  appendMessage: ({ conversationId, message }: AppendMessageProps) => Message[];
-  amendMessage: ({ conversationId, content }: AmendMessageProps) => void;
+  appendMessage: ({
+    conversationId,
+    message,
+  }: AppendMessageProps) => Promise<Message[] | undefined>;
+  amendMessage: ({ conversationId, content }: AmendMessageProps) => Promise<void>;
   appendReplacements: ({
     conversationId,
     replacements,
-  }: AppendReplacementsProps) => Record<string, string>;
-  clearConversation: (conversationId: string) => void;
-  createConversation: ({ conversationId, messages }: CreateConversationProps) => Conversation;
+  }: AppendReplacementsProps) => Promise<Record<string, string> | undefined>;
+  clearConversation: (conversationId: string) => Promise<void>;
+  getDefaultConversation: ({ conversationId, messages }: CreateConversationProps) => Conversation;
   deleteConversation: (conversationId: string) => void;
-  removeLastMessage: (conversationId: string) => Message[];
-  setApiConfig: ({ conversationId, apiConfig }: SetApiConfigProps) => void;
-  setConversation: ({ conversation }: SetConversationProps) => void;
+  removeLastMessage: (conversationId: string) => Promise<Message[] | undefined>;
+  setApiConfig: ({
+    conversation,
+    apiConfig,
+  }: SetApiConfigProps) => Promise<Conversation | IHttpFetchError<unknown>>;
+  createConversation: (conversation: Conversation) => Promise<Conversation | undefined>;
+  getConversation: (conversationId: string) => Promise<Conversation | undefined>;
 }
 
 export const useConversation = (): UseConversation => {
   const {
     allSystemPrompts,
     assistantTelemetry,
     knowledgeBase: { isEnabledKnowledgeBase, isEnabledRAGAlerts },
-    setConversations,
+    http,
   } = useAssistantContext();
 
+  const getConversation = useCallback(
+    async (conversationId: string) => {
+      const currentConversation = await getConversationById({ http, id: conversationId });
+      if (isHttpFetchError(currentConversation)) {
+        return;
```

Comment:
Consider displaying an error toast indicating the conversation cannot be retrieved